### PR TITLE
appyconfig: support symlinks

### PR DESCRIPTION
**Pull request by hongkailiu**
Required by https://github.com/openshift/release/pull/10545

/cc @alvaroaleman

### CODE REVIEW

File: cmd/applyconfig/applyconfig.go [hunk 1]
```diff
@@ -208,6 +208,17 @@ func applyConfig(rootDir string, o *options) error {
 			return err
 		}
 
+		if info.Mode()&os.ModeSymlink != 0 {
+			target, err := os.Readlink(path)
+			if err != nil {
+				return err
+			}
+			logrus.Infof("replace the symlink %s with the target %s", path, target)
+			if err := applyConfig(target, o); err != nil {
```
Comment (alvaroaleman) [inline cmd/applyconfig/applyconfig.go:217]: Looks correct. Can we get a test for this?

Comment (hongkailiu) [reply cmd/applyconfig/applyconfig.go]: `func applyConfig(rootDir string, o *options) error` is not very unit test friendly: it does not give me a chance to inject the executor.

Instead, i tested the bin:

```
➜  test.config ll ./lll/
total 0
-rw-r--r--  1 hongkliu  staff     0B 29 Jul 14:56 abc.txt
-rw-r--r--  1 hongkliu  staff     0B 29 Jul 14:56 abc.yaml
➜  test.config tree .
.
├── lll -> ../lll
└── ttt.yaml -> ../ttt.yaml

1 directory, 1 file
```

```
applyconfig --config-dir ./ --kubeconfig ~/.kube/config --as=
INFO[0000] Applying config in directory: ./
INFO[0000] replace the symlink folder lll with the target ../lll
INFO[0000] Applying config in directory: ../lll
ERRO[0000] oc apply -f ../lll/abc.yaml --kubeconfig /Users/hongkliu/.kube/config --dry-run: failed to apply
W0729 16:11:16.833044   37986 helpers.go:535] --dry-run is deprecated and can be replaced with --dry-run=client.
error: no objects passed to apply
INFO[0000] Ran apply command                             output=
ERRO[0001] oc apply -f ttt.yaml --kubeconfig /Users/hongkliu/.kube/config --dry-run: failed to apply
W0729 16:11:17.565517   37992 helpers.go:535] --dry-run is deprecated and can be replaced with --dry-run=client.
error: no objects passed to apply
INFO[0001] Ran apply command                             output=
ERRO[0001] There were failures while applying config     error="failed to apply config"
```

**Comment by stevekuznetsov** [comment]
Looks good by inspection

**Pull request merged**